base_classes/NXdata.nxdl.xml

```diff
@@ -85,124 +85,170 @@
-			  y: float[10,20]     --> coordinates along both dimensions
-
-		In this example each data point ``data[i,j]`` has axis coordinates ``[x[i,j], y[i,j]]``.
+			  @eta_indices = [0]
+			  @tth_indices = [1]
+			  @q_indices = [1]
+			  eta: float[10]    --> coordinates along the first dimension
+			  tth: float[20]    --> coordinates along the second dimension (alternative)
+			  q: float[20]      --> coordinates along the second dimension (default)
 
-		`Dimensions`
+		Note that to recognize ``tth`` as an axis, ``@tth_indices`` must be present. Readers that do
+		not make use of the :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attributes will
+		not understand that ``tth`` is an axis and cannot take this field into account.
 
-		The data dimensions annotated by an :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field are defined by the
-		:ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attribute. When this attribute is missing,
-		the position(s) of the :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` string in the
-		:ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute are used.
+		``3. Multi-dimensional axes``
 
-		When all :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields are one-dimensional, and none of the data dimensions
-		have more than one axis, the :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attributes
-		are often omitted. If one of the data dimensions has no :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field,
-		the string “.” can be used in the corresponding index of the axes list.
+		When coordinates do not form a regular grid, multi-dimensional :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields
+		can be used. Just remember that the shape of an :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field must be equal to the shape
+		of the :ref:`DATA &lt;/NXdata/DATA-field&gt;` dimensions it spans.
 
-		An example of this would be
+		An example of this would be 2D scatter data where each data point has unique coordinates
 
 		.. code-block::
 
 			data:NXdata
 			  @signal = "data"
-			  @axes = ["x", ".", "z"]  --> the order matters
-			  data: float[10,20,30]
-			  x: float[10]             --> coordinates along the first dimension
-			  z: float[30]             --> coordinates along the third dimension
+			  @axes = ["x", "y"]
+			  @x_indices = [0, 1]
+			  @y_indices = [0, 1]
+			  data: float[10,20]
+			  x: float[10,20]     --> coordinates along both dimensions
+			  y: float[10,20]     --> coordinates along both dimensions
 
-		When using :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` this becomes
+		In this example each data point ``data[i,j]`` has axis coordinates ``[x[i,j], y[i,j]]`` and when
+		plotting, ``x`` is used along the first data dimension by default and `y` along the second data dimension.
+		Since ``x`` and ``y`` span both dimensions, a reader could choose to use ``y`` for the first dimension
```

Comment:
I think it's the user who chooses - the reader may allow a choice but it should obey the default order given by `@axes`